druid/src/widget/flex.rs

```diff
@@ -659,54 +657,109 @@ impl CrossAxisAlignment {
     fn align(self, val: f64) -> f64 {
         match self {
             CrossAxisAlignment::Start => 0.0,
-            CrossAxisAlignment::Center => val / 2.0,
+            CrossAxisAlignment::Center => (val / 2.0).round(),
             CrossAxisAlignment::End => val,
         }
     }
 }
 
 impl MainAxisAlignment {
-    fn spacing(self, extra: f64, n_children: usize) -> Spacing {
+    /// Given the provided extra space and children count,
+    /// this returns a `Vec<f64>` of spacing,
+    /// where the first element is the spacing before any children
+    /// and all subsequent elements are the spacing after children.
+    ///
+    /// If the extra space is infinite then it returns an empty `Vec<f64>`.
+    fn spacing(self, extra: f64, n_children: usize) -> Vec<f64> {
```

Comment:
An observation: this feels like something that could be very naturally expressed as an iterator. This isn't a blocker or anything, but it feels like it could be a nice win.

Reply:
I'll play around with `Iterator` a bit and see if I can construct code that is more readable.

Reply:
Okay, I pushed the iterator rewrite.